### PR TITLE
🌱 hack/observability: skip preload images on non-kind clusters

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Fixes #

### CODE REVIEW

File: hack/tools/tilt-prepare/main.go [hunk 1]
```diff
@@ -283,9 +287,11 @@ func tiltResources(ctx context.Context, ts *tiltSettings) error {
 	// NOTE: strictly speaking cert-manager is not a resource, however it is a dependency for most of the actual resources
 	// and running this is the same task group of the kustomize/provider tasks gives the maximum benefits in terms of reducing the total elapsed time.
 	if ts.DeployCertManager == nil || *ts.DeployCertManager {
-		tasks["cert-manager-cainjector"] = preLoadImageTask(fmt.Sprintf("quay.io/jetstack/cert-manager-cainjector:%s", config.CertManagerDefaultVersion))
-		tasks["cert-manager-webhook"] = preLoadImageTask(fmt.Sprintf("quay.io/jetstack/cert-manager-webhook:%s", config.CertManagerDefaultVersion))
-		tasks["cert-manager-controller"] = preLoadImageTask(fmt.Sprintf("quay.io/jetstack/cert-manager-controller:%s", config.CertManagerDefaultVersion))
+		if ts.PreloadCertManagerImages == nil || *ts.PreloadCertManagerImages {
```
Comment (killianmuldoon) [inline hack/tools/tilt-prepare/main.go:290]: Fine with the change in general - but this 'defaulting to true` setup seems kind of weird. Could we clean this up by having a defaulting function for tiltSettings?

**Review by killianmuldoon — COMMENTED** [review]
Looks good - one nit

File: hack/tools/tilt-prepare/main.go (hunk 1 above)
Comment (sbueringer) [reply hack/tools/tilt-prepare/main.go:290]: We have the defaulting function (I also adjusted it). This is just double-checking that we don't panic. The same we have for DeployCertManager.

I would be fine with dropping the nil check in both cases

Comment (killianmuldoon) [reply hack/tools/tilt-prepare/main.go:290]: Yeah - it's just slightly confusing. I guess the alternative would be to default to a struct with no pointers, but it's immaterial for this PR.

File: docs/book/src/developer/tilt.md [hunk 2]
```diff
@@ -276,6 +276,9 @@ Supported settings:
 
 **deploy_cert_manager** (Boolean, default=`true`): Deploys cert-manager into the cluster for use for webhook registration.
 
+**preload_cert_manager_images** (Boolean, default=`true`): Preloads cert-manager images into the cluster. This allows disabling preloading images,
```
[the remark this answers is not in the document]
Comment (sbueringer) [reply docs/book/src/developer/tilt.md:279]: Sounds good to me

File: hack/tools/tilt-prepare/main.go (hunk 1 above)
Comment (sbueringer) [reply hack/tools/tilt-prepare/main.go]: Got rid of the new flag :D

File: docs/book/src/developer/tilt.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sbueringer) [reply docs/book/src/developer/tilt.md]: Done

**Comment by sbueringer** [comment]
@fabriziopandini @killianmuldoon PTAL :)

(cc @ykakarap pushed the same version on the scale branch, so the flag is not needed anymore)

**Review by killianmuldoon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**